consensus/src/lib.rs

```diff
@@ -174,5 +174,23 @@ cfg_if::cfg_if! {
             /// may change (and old shares may be deleted).
             fn share(&self, index: Self::Index) -> Option<&Self::Share>;
         }
+
+        /// Monitor is the interface an external actor can use to observe the progress of a consensus implementation.
+        ///
+        /// Monitor is used to implement mechanisms that share the same set of active participants as consensus and/or
+        /// perform some activity that requires some synchronization with the progress of consensus.
+        ///
+        /// Monitor can be implemented using [`Committer`](crate::Committer) to avoid introducing complexity
+        /// into any particular consensus implementation.
+        pub trait Monitor: Clone + Send + 'static {
```

Comment:
I considered just having `ordered_broadcast` implement `Committer` (and use that) but I figured there may be multiple consumers of that stream (and/or some implementation may prefer to expose different types of monitors directly). Open to feedback on this decision.